core/src/main/java/nu/yona/server/analysis/service/AnalysisEngineService.java

```diff
@@ -74,24 +83,33 @@ private void addOrUpdateActivity(PotentialConflictDTO payload, UserAnonymizedDTO
 			}
 		}
 		// Update message only if it is within five seconds to avoid unnecessary cache flushes.
-		else if (now.getTime() - activity.getEndTime().getTime() >= yonaProperties.getAnalysisService().getUpdateSkipWindow())
+		else if (payload.endTime.getTime() - activity.getEndTime().getTime() >= yonaProperties.getAnalysisService()
+				.getUpdateSkipWindow())
 		{
-			updateLastActivity(payload, now, matchingGoal, activity);
+			updateLastActivity(payload, userAnonymized, matchingGoal, activity);
 			cacheService.updateLatestActivityForUser(activity);
 		}
+		else if (payload.startTime.before(activity.getStartTime()))
+		{
+			// can happen if app activity is posted after some network activity is posted during the use of the same app
+			// TODO: remove activities after start time and log a new activity?
+			// or just overwrite start time of the existing activity, creating possible overlap?
```

Comment:
What could go wrong if we just update the start time? If that's nothing or nothing dramatic, than I'd go for that.